### PR TITLE
GIBCT Search: University of Maryland institutions dont show in search results when searching "University of Maryland" #13408

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/13408

- new column to weams and institutions called institution_search that gets the same processing as the search term when using the trigram % operator
- add sort weight for institution starts with cases
- add sort weight for institution_search similarity to processed search term
- add sort weight with a modifier for when ialias contains the search term as a word but not within words
- remove sort weight for ialias similarity
- fix common word removal, was removing common words from within words
- set `pg_tgrm. similarity_threshold` to 0.4

## Testing done
In CT search using the following terms:
- "college of charlesotn" for misspelling
- "college of charleston" for exact match
- "university of maryland" for starts with/contains
- "mit" for alias
- "bbc" for alias

## Screenshots
Screen Shot 2020-09-10 at 5 23 02 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/backend-review-group this is ready for review

First notification was accidental

File: app/models/institution.rb [hunk 1]
```diff
@@ -228,10 +228,16 @@ def self.autocomplete(search_term, limit = 6)
...
+  # Idea is to have a processed version of the institution column available to compare with
+  # the trigram % operator against the processed search term
   def self.institution_search_term(search_term)
+    return if search_term.blank?
+
     processed_search_term = search_term.clone
     Settings.search.common_word_list.each do |word|
-      processed_search_term = processed_search_term.gsub(Regexp.new(word, Regexp::IGNORECASE), '')
+      processed_search_term = processed_search_term.gsub(/\b#{Regexp.escape(word)}\b/i, '') if word.match(/[a-z]/i)
+      processed_search_term = processed_search_term.gsub(/#{Regexp.escape(word)}/i, '')
```
Comment (erluti) [inline app/models/institution.rb:240]: these could be `gsub!` instead of assigning the variable each time

**Review by philipwritescode — APPROVED** [review]
LGTM, I chuckled at "MUST INVESTIGATE TARANTULAS"

File: app/models/institution.rb (hunk 1 above)
Comment (zurbergram) [reply app/models/institution.rb]: changed, had to switch `clone` to `dup` to use

**Pull request merged**